.github/workflows/ci-build.yml

```diff
@@ -35,4 +35,17 @@ jobs:
         run: go install -ldflags "-X github.com/openziti/zrok/build.Version=${{ github.ref }} -X github.com/openziti/zrok/build.Hash=${{ github.sha }}" ./...
 
       - name: test
-        run: go test -v ./...
\ No newline at end of file
+        run: go test -v ./...
+
+      - name: solve GOBIN
+        id: solve_go_bin
+        run: |
+          echo DEBUG: go_path="$(go env GOPATH)"
+          echo go_bin="$(go env GOPATH)/bin" >> $GITHUB_OUTPUT
+
+      - name: upload build artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: linux-amd64
```

Comment:
And we're only solving for `amd64`?

Reply:
The CI build artifact is stored in a temporary GitHub cache where it can be downloaded from the build job's summary page. [Here's an example job page for a Ziti build with downloadable artifacts built from the branch](https://github.com/openziti/ziti/actions/runs/4555833471).

Presently, we're only testing the build for `linux-amd64`, so that is the only artifact available for upload and download on branches. If there's a need to test the build for other variants or to conveniently download artifacts for other platform variants, or both, then we can add those variants to this CI job and make them available in the same temporary storage location for convenient download by any GitHub user.